chia/full_node/block_store.py

```diff
@@ -481,48 +458,43 @@ async def set_peak(self, header_hash: bytes32) -> None:
 
         if self.db_wrapper.db_version == 2:
             # Note: we use the key field as 0 just to ensure all inserts replace the existing row
-            cursor = await self.db.execute("INSERT OR REPLACE INTO current_peak VALUES(?, ?)", (0, header_hash))
-            await cursor.close()
+            await self.db.execute("INSERT OR REPLACE INTO current_peak VALUES(?, ?)", (0, header_hash))
         else:
-            cursor_1 = await self.db.execute("UPDATE block_records SET is_peak=0 WHERE is_peak=1")
-            await cursor_1.close()
-            cursor_2 = await self.db.execute(
+            await self.db.execute("UPDATE block_records SET is_peak=0 WHERE is_peak=1")
+            await self.db.execute(
                 "UPDATE block_records SET is_peak=1 WHERE header_hash=?",
                 (self.maybe_to_hex(header_hash),),
             )
-            await cursor_2.close()
 
     async def is_fully_compactified(self, header_hash: bytes32) -> Optional[bool]:
-        cursor = await self.db.execute(
+        async with self.db.execute(
             "SELECT is_fully_compactified from full_blocks WHERE header_hash=?", (self.maybe_to_hex(header_hash),)
-        )
-        row = await cursor.fetchone()
-        await cursor.close()
+        ) as cursor:
+            row = await cursor.fetchone()
         if row is None:
             return None
         return bool(row[0])
 
     async def get_random_not_compactified(self, number: int) -> List[int]:
 
         if self.db_wrapper.db_version == 2:
-            cursor = await self.db.execute(
+            async with self.db.execute(
                 f"SELECT height FROM full_blocks WHERE in_main_chain=1 AND is_fully_compactified=0 "
                 f"ORDER BY RANDOM() LIMIT {number}"
-            )
+            ) as cursor:
+                rows = await cursor.fetchall()
         else:
             # Since orphan blocks do not get compactified, we need to check whether all blocks with a
             # certain height are not compact. And if we do have compact orphan blocks, then all that
             # happens is that the occasional chain block stays uncompact - not ideal, but harmless.
-            cursor = await self.db.execute(
+            async with self.db.execute(
                 f"SELECT height FROM full_blocks GROUP BY height HAVING sum(is_fully_compactified)=0 "
                 f"ORDER BY RANDOM() LIMIT {number}"
-            )
-
-        rows = await cursor.fetchall()
-        await cursor.close()
+            ) as cursor:
+                rows = await cursor.fetchall()
 
-        heights = []
-        for row in rows:
-            heights.append(int(row[0]))
+            heights = []
+            for row in rows:
+                heights.append(int(row[0]))
```

Comment:
https://github.com/Chia-Network/chia-blockchain/runs/4571820373?check_suite_focus=true

```suggestion
        heights = [int(row[0]) for row in rows]
```